packages/shopping-cart/src/action-promise-error.ts

```diff
@@ -0,0 +1,9 @@
+export class ActionPromiseError extends Error {
```

Comment:
I would call it `CartActionError`. Such a name self-explains that the error is related to the cart, and leaves out the irrelevant fact that it's typically used for rejected promises. A sync operation could throw it as well.